docs/book/src/tasks/experimental-features/runtime-sdk/implement-topology-mutation-hook.md

```diff
@@ -192,6 +296,97 @@ function openSwaggerUI() {
 }
 </script>
 
+### DiscoverVariables
+
+A DiscoverVariables call returns definitions for one or more variables. This variable definition is the same as that used for inline variables.
+
+Example variable definition:
+
+```yaml
+  - name: etcdImageTag # name of the variable
+    required: true # whether the variable must be defined for Clusters.
+    schema:
+      openAPIV3Schema: 
+        type: string 
+        default: "3.5.3-0" # variable value which is defaulted during reconciliation if not defined elsewhere.
+        example: "3.5.3-0" # example for user guidance.
+        description: "etcdImageTag sets the tag for the etcd image." #description of variable use.
+        # Additional fields from clusterv1.JSONSchemaProps
+```
+
+#### Example Request:
+
+* The request is a simple request which contains the name and namespace of the ClusterClass it is called for.
+TODO: Should this send the full ClusterClass?
```

Comment:
I would start simple. I probably wouldn't even send the ClusterClass. I'm not sure if we should facilitate that folks are coupling their external patch extension to closely to a specific ClusterClass (either by name or by get'ing the ClusterClass - which can always be out of sync).

If we don't have a use case for folks accesing the ClusterClass from the external patch extension I wouldn't add any fields for now.

They can still do some parameterization via settings.